src/workerd/api/streams/internal.h

```diff
@@ -143,6 +147,10 @@ class ReadableStreamInternalController: public ReadableStreamController {
   bool disturbed = false;
   bool readPending = false;
 
+  // Used by Sockets code to signal to the ReadableStream that it should error when read from
+  // because the socket is currently being closed.
+  bool errorOnReadsDueToPendingClosure = false;
```

Comment:
Other idea is to stick it into the StreamState but there seems to be a bunch of code that checks for `!StreamState.is<Unlocked>` and I didn't want to mess with it.